metadatastore/commands.py

```diff
@@ -638,6 +638,29 @@ def find_events(**kwargs):
 
 
 @_ensure_connection
+def count_events(**kwargs):
+    """Like find_events, but it returns the count, not the actual Events.
+
+    This function is likely to be deprecated in this near future. Instead,
+    find_events will return an iterator that has a length.
+    """
+    # Some user-friendly error messages for an easy mistake to make
+    if 'event_descriptor' in kwargs:
+        raise ValueError("Use 'descriptor', not 'event_descriptor'.")
+    if 'event_descriptor_id' in kwargs:
+        raise ValueError("Use 'descriptor_id', not 'event_descriptor_id'.")
+
+    _format_time(kwargs)
+    try:
+        kwargs['descriptor_id'] = kwargs.pop('descriptor').id
+    except KeyError:
+        pass
+    _normalize_object_id(kwargs, '_id')
+    _normalize_object_id(kwargs, 'descriptor_id')
+    return len(Event.objects(__raw__=kwargs))
```

Comment:
I suspect that `Event.objects(__raw__=kwargs).count()` will be faster as it will skip the making all of the mongoengine objects step (which were the source of the major performance hit at HXN).